src/main/java/org/jenkinsci/plugins/github/webhook/GHSignatureHeader.java

```diff
@@ -0,0 +1,50 @@
+package org.jenkinsci.plugins.github.webhook;
+
+import org.kohsuke.stapler.AnnotationHandler;
+import org.kohsuke.stapler.InjectedParameter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.slf4j.Logger;
+
+import javax.servlet.ServletException;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * InjectedParameter annotation to use on WebMethod parameters.
+ * Grabs the X-Hub-Signature header value from HTTP requests and injects them to function parameters.
+ *
+ * @author lanwen (Merkushev Kirill)
```

Comment:
My bad, the code was mostly a copy-paste of a class he had written. I should be more explicit and say it was based off code he had written.